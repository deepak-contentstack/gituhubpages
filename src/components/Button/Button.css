@import '../../../tokens/variables.scss';
@import '../../../tokens/tokens.css';
@import '../../static/styles/css/typography.css';
@import '../../static/styles/css/common.css';

[type='button']:not(:disabled),
[type='reset']:not(:disabled),
[type='submit']:not(:disabled),
button:not(:disabled) {
  	cursor: pointer;
}

.ui-component button{
  	margin: 0 calc(var(--space-2) + 1px);
}

/* Button */

.Button {
	background-color: transparent;
	border: 1px solid transparent;
	border-radius: $radii-6;
	font-family: $font-family-primary;
	font-size: $size-font-medium;
	font-weight: $font-weight-bold;
	line-height: $space-12;
	/* line-height: calc(var(--space-12) + 1px); */
	min-height: $space-32;
	min-width: $space-32;
	padding: $space-8 $space-16;
	/* padding: $space-8 $space-16 calc(var(--space-8) + 1px); */
	position: relative;
	text-align: center;
	transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
	border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
	vertical-align: middle;

	&__loader{
		left: 50%;
		position: absolute;
		top: 50%;
		-webkit-transform: translateX(-50%) translateY(-50%);
		-moz-transform: translateX(-50%) translateY(-50%);
	}

	&__icon {
		align-items: center;
		color: $color-base-gray-20;
		display: flex;
		height: 100%;
		justify-content: center;
		left: 0;
		position: absolute;
		top: 0;
		transition: border-color 0.15s ease-in-out;
		width: 2rem;
	}

	&.Button--icon {
		padding: 0;
		svg {
			margin : auto!important;
		}
	}

	&.Button--icon-alignment-right{
		svg:first-child{
		float: right;
		margin-left: 0.625rem;
		margin-right: 0;
		}
	}

	&:focus {
		box-shadow: 0px 0px 0px 4px #BBB4F4;
		outline: none;
	}

	&--state {
		&-disabled {
			background-color: $color-brand-secondary-lighter !important;
			border: 1px solid transparent;
			color: $color-font-tertiary !important;
			opacity: $opacity-5;
			pointer-events: none;
			text-shadow: none !important;
			& svg path {
				fill: $color-font-tertiary !important
			}
		}
		&-loading{
			cursor: default;
			pointer-events: none;
		}
	}

	&--style {
		&-circle {
			border-radius: $radii-50;
			padding: 0;
		}

		&-fullwidth {
			width: 100%;
		}
	}

	&--primary {
		background-color: $color-brand-primary-base;
		color: $color-font-white;
		text-shadow: 0 1px 0 rgba(0, 0, 0, 0.16);

		&:hover {
			background-color: $color-brand-primary-darker;
		}

		&:focus {
			box-shadow: 0px 0px 0px 4px #BBB4F4;
		}

		&:active {
			background-color: $color-brand-primary-dark;
		}

		svg:first-child {
			margin: 0 $space-10 0 0;
			vertical-align: top;
		}
	}

	&--secondary {
		background-color: $color-brand-white-base;
		border: 1px solid $color-border-link;
		color: $color-font-link;
		
		&:hover{
			border:1px solid $color-base-purple-25;
			color:$color-base-purple-25;
		}

		svg:first-child {
			margin: 0 $space-10 0 0;
			vertical-align: top;
		}

		&:hover {
			border : 1px solid $color-base-purple-25;
			color: $color-brand-primary-darker;
		}

		&:focus {
			box-shadow: 0px 0px 0px 4px #BBB4F4;
		}

		&:active {
			border: 1px solid $color-brand-primary-dark;
			color: $color-brand-primary-dark;
		}
	}

	&--tertiary {
		color: $color-font-link;
		font-size: $size-font-large;
		font-weight: $font-weight-regular;
		
		svg:first-child {
			margin: 0 $space-10 0 0;
			vertical-align: top;
			path {
				fill: $color-font-link;
			}
		}
	}

	&--tertiary-outline {
		background: $color-brand-white-base;
		border: .0625rem solid $color-base-gray-5;
		color: $color-font-base;
		font-size: $size-font-large;
		font-weight: $font-weight-regular;
		height: 2.5rem;
		padding: .4375rem .625rem;

		&:hover {
			border: .0625rem solid $color-border-link;
			color: $color-font-link;

			svg:first-child {
				path, circle, rect {
					fill: $color-font-link;
					stroke: $color-font-link;
				}
			}
		}
		
		svg:first-child {
			margin: 0 $space-10 0 0;
			vertical-align: top;
		} 
	}

	&--outline {
		background: $color-brand-white-base;
		border: 1px solid $color-border-dark;
		color: $color-font-base;
		font-size: $size-font-large;
		font-weight: $font-weight-regular;

		&:hover {
			border: 1px solid $color-border-link;
			color: $color-font-link;
		}
	}

	&--success {
		background-color: $color-brand-white-base;
		color: $color-font-success;
		&:hover {
			background-color: $color-brand-success-base;
			border: 1px solid $color-base-white-5;
			color: $color-font-white;
		}
	}

	&--control {
		border: 1px solid $color-base-gray-5;
		border-radius: $radii-8;
		color: $color-base-gray-20;
		font-size: $size-font-large;
		font-weight: $font-weight-regular;
		line-height: 1;
		max-width: 11.563rem;
		overflow: hidden;
		padding: $space-12 $space-12 $space-12 $space-30;
		position: relative;
		text-overflow: ellipsis;
		white-space: nowrap;

		&:hover {
			border-color: $color-brand-primary-base;
			color: $color-brand-primary-base;
			
			svg {
				path {
				stroke: $color-brand-primary-base;
				}
			}
		}
	}

	&--delete {
		border: 1px solid $color-border-warning;
		color: $color-font-warning;
		svg {
			margin: 0 $space-10 0 0;
			vertical-align: top;
			path {
				fill : $color-base-orange-5;
			}
		}
		&:hover {
			background-color: $color-brand-warning-base;
			color: $color-font-white;
			svg {
				path {
					fill : $color-base-white-5;
				}
			}
		}
	}

	&--outline-delete {
		background-color: $color-brand-white-base;
		border: 1px solid $color-border-link;
		color: $color-font-link;
		svg {
			margin: 0 .625rem 0 0;
			path {
				fill : $color-base-purple-5;
			}
		}
		&:hover{
			background-color:$color-brand-warning-base;
			border: 1px solid $color-border-warning;
			color: $color-font-white;
			svg {
				path {
					fill : $color-base-white-5;
				}
			}
		}
	}

	&--outline-success {
		background-color: $color-brand-white-base;
		border: 1px solid $color-border-success;
		color: $color-font-success;
		svg {
			margin: 0 .625rem 0 0;
			vertical-align: bottom;
		}
		&:hover {
			background-color: $color-brand-success-base;
			color: $color-font-white;
			svg {
				path {
					fill : $color-base-white-5;
					stroke: $color-base-white-5;
				}
			}
		}
	}
	
	&--bulk-delete{
		background-color: $color-brand-primary-base;
		color: $color-font-white;
		svg {
			margin: 0 .625rem 0 0;
			path {
				fill : $color-font-white;
			}
		}
		&:hover{
			background-color: $color-font-white;
			color: $color-base-orange-5;
			svg {
				path {
					fill : $color-base-orange-5;
				}
			}
		}
	}

	&--danger {
		background-color: $color-brand-white-base;
		color: $color-font-warning;
		&:hover {
			background-color: $color-brand-warning-base;
			border: 0.0625rem solid $color-base-white-5;
			color: $color-font-white;
		}
	}

	&--warning {
		background-color: $color-brand-white-base;
		color: $color-font-attention;
		&:hover {
			background-color: $color-brand-attention-base;
			border: 0.0625rem solid $color-base-white-5;
			color: $color-font-white;
		}

		&-link {
			color: $color-base-purple-5;
			font-weight: $font-weight-regular;
			text-decoration: none;
		}
	}

	&--light {
		color: $color-brand-secondary-base;
		&:hover {
			background-color: $color-brand-secondary-lighter;
			color: $color-font-base;
		}
	}

	&--white {
		background: $color-brand-white-base;
		box-shadow: 0 8px 8px rgba(0, 0, 0, 0.02), 0 4px 8px rgba(0, 0, 0, 0.06), 0 8px 16px rgba(0, 0, 0, 0.04);
		color: $color-font-secondary;
		font-weight: $font-weight-regular;
		svg {
			margin-right: $space-10;
		}
		&:hover {
			background: $color-brand-secondary-lighter;
			color: $color-font-black;
			svg path {
				fill : $color-base-gray-20;
			}
		}
	}

	&--size {
		&-large {
			font-size: $size-font-xl;
			padding: $space-12 $space-18 $space-12;
		}

		&-small {
			font-size: $size-font-medium;
			min-height: $space-25;
			padding: $space-5 $space-10;
		}

		&-fullwidth {
			width: 100%;
		}
	}

	&:active {
		border: 1px solid $color-brand-primary-dark;
	}

	.badge {
		margin-left: $space-10;

		&:first-child {
			margin: 0;
			margin-right: $space-10;
		}
	}
}

[type='button']:not(:disabled),
[type='reset']:not(:disabled),
[type='submit']:not(:disabled),
button:not(:disabled) {
  	cursor: pointer;
}

.Button-wrapper button {
  	margin: 0 calc(var(--space-2) + 1px);
}

.Button-container {
	height: 6.25rem;
	margin: $space-15;
	padding: $space-34 0;
	text-align: center;
	width: 12.5rem;
}
