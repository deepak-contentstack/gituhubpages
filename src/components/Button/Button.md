# Button

## Best practices

- Use clear and accurate labels.
- Start button labels with verbs, for example "Add field", not "New field".
- Keep the position of the buttons consistent on the interface.
- Reduce the complexity by using a small number of actions on the interface. It may create confusion if too many actions are used.
