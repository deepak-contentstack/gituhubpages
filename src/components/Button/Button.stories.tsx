import React from 'react';
import { action } from '@storybook/addon-actions';

import { Button } from '../../index'
import DocSubData from '../../utils/components/DocSubData';

import notes from './Button.md'

const buttonTypeArray: any = ['primary', 'secondary', 'tertiary', 'warning', 'danger', 'success', 'light', 'control', 'delete', 'white', 'bulk-delete', 'outline', 'tertiary-outline', 'outline-delete', 'outline-success']

const iconArray: any = ['PublishAsset', 'Upload', 'Send', 'Proceed'];

const iconAlignmentArray: any = ['left', 'right'];

const buttonSize: any = ['small', 'large', 'default'];

const typeArray: any = ['button', 'submit', 'reset'];

const componentDescription = 'The Button component lets you communicate an action that needs to occur when you click on them. They contain a text label to describe the action, and an icon if appropriate.'

const usageDescription = 'On clicking a button, an event is triggered that initiates an action attached to it. Buttons can contain letters, icons, or a label to describe the action that is performed.'

export default {
  title: 'Components / Button',
  component: Button,
  argTypes: {
    label: {
      control: { type: 'text' },
      description: 'Provide a label to your button',
      table: {
        type: {
          summary: 'text'
        }
      }
    },
    isFullWidth: {
      control: { type: 'boolean' },
      description: 'If button covers full width',
      table: {
        type: {
          summary: 'boolean'
        },
        defaultValue: {
          summary: false
        }
      }
    },
    // isRounded: {
    //   control: { type: 'boolean' },
    //   description: 'If button is rounded',
    //   table: {
    //     type: {
    //       summary: 'boolean'
    //     },
    //     defaultValue: {
    //       summary: false
    //     }
    //   }
    // },
    size: {
      control: { type: 'select', options: buttonSize },
      description: 'Determine the size of the button: small or large',
      table: {
        type: {
          summary: 'select'
        }
      }
    },
    disabled: {
      control: { type: 'boolean' },
      description: 'Determines whether button is disabled or not',
      table: {
        type: {
          summary: 'boolean'
        },
        defaultValue: {
          summary: false
        }
      }
    },
    // type: {
    //   control: { type: 'select', options: typeArray },
    //   description: 'Type of button action (button, submit,reset)',
    //   table: {
    //     type: {
    //       summary: 'select'
    //     },
    //     defaultValue: {
    //       summary: 'button'
    //     }
    //   }
    // },
    className: {
      control: { type: 'disabled' },
      description: 'Provide the class names to be appended to this prop',
      table: {
        type: {
          summary: 'string'
        }
      }
    },
    buttonType: {
      control: { type: 'select', options: buttonTypeArray },
      description: 'Determine the type of button you want to use',
      table: {
        type: {
          summary: 'select'
        },
        defaultValue: {
          summary: 'primary'
        }
      }
    },
    icon: {
      control: { type: 'select', options: iconArray },
      description: 'Pass the icon that you want to add in the button',
      table: {
        type: {
          summary: 'select'
        }
      }
    },
    iconAlignment: {
      control: { type: 'select', options: iconAlignmentArray },
      description: 'Determines the alignment of the icon',
      table: {
        type: {
          summary: 'select'
        }
      }
    },
    isLoading: {
      control: { type: 'boolean' },
      description: 'Used to set the button in the loading state',
      table: {
        type: {
          summary: 'boolean'
        },
        defaultValue: {
          summary: false
        }
      }
    },
    loadingColor: {
      control: { type: 'color' },
      description: 'Determines the color of the loading animation you want to display',
      table: {
        type: {
          summary: 'color'
        }
      }
    },
    onClick: {
      control: { type: 'disabled' },
      description: 'Action that you want to perform when a button is clicked',
      table: {
        type: {
          summary: 'MouseEventHandle'
        }
      }
    },
    children: {
      control: { type: 'disabled' },
      description: 'Action that you want to perform when a button is clicked',
      table: {
        type: {
          summary: 'ReactReactNode'
        }
      }
    },
    href: {
      control: { type: 'disabled' },
      description: 'If you want a button as a link, provide the details in href.',
      table: {
        type: {
          summary: 'string'
        }
      }
    },
    style: {
      control: { type: 'disabled' },
      description: 'If you want a button as a link, provide the details in href.',
      table: {
        type: {
          summary: 'CSSProperties'
        }
      }
    },
    onlyIcon: {
      control: { type: 'boolean' },
      description: 'If you want to show only icon, set the param to ‘true’',
      table: {
        type: {
          summary: 'boolean'
        },
        defaultValue: {
          summary: false
        }
      }
    },
    iconProps: {
      control: { type: 'diabled' },
      description: 'Pass additional props for icon',
      table: {
        type: {
          summary: 'Partial'
        }
      }
    },
    id: {
      control: { type: 'diabled' },
      description: 'Pass id to the button ',
      table: {
        type: {
          summary: 'string'
        }
      },
    },
    testId: {
      control: { type: 'diabled' },
      description: 'An ID used for testing purposes applied as a data attribute (data-test-id)',
      table: {
        type: {
          summary: 'string'
        },
        defaultValue: {
          summary: 'cs-button'
        }
      },
    }
  },
  parameters: {
    componentSubtitle: <DocSubData componentDescription={componentDescription} usageDescription={usageDescription} />,
    notes: { notes },
    design: {
      type: 'figma',
      url:
        'https://www.figma.com/proto/ovlEsic8nr81y5BOu1OlqD/ui-inventory-v2?node-id=22%3A3&scaling=min-zoom',
    }
  },
}

const tableDisable = { table: { disable: true } }

export const Default = (args) => {
  return (
    <Button isFullWidth={args.isFullWidth} style={{ backgroundColor: "brown" }} isRounded={args.isRounded} size={args.size} disabled={args.disabled} buttonType={args.buttonType} iconAlignment={args.iconAlignment} isLoading={args.isLoading} loadingColor={args.loadingColor} onlyIcon={args.onlyIcon} icon={args.onlyIcon ? 'Send' : args.icon} onClick={action('onClick')}>{args.onlyIcon ? '' : args.label}</Button>
  );
};
Default.args = { label: 'Default', onlyIcon: false };

export const TypesOfButtons = (args) => {

  return (
    <Button onClick={action('onClick')} buttonType={args.buttonType}>{args.label}</Button>
  );
}

TypesOfButtons.args = { label: 'Types of buttons', buttonType: 'primary' };
TypesOfButtons.parameters = {
  docs: {
    storyDescription: `We have multiple button types such as primary, secondary, tertiary, and so on based on various user requirements. To select a button type, go to <b>Controls</b> in the <b>Canvas</b> area, and click on <b>buttonType</b> dropdown.`
  }
}
TypesOfButtons.argTypes = {
  isFullWidth: tableDisable,
  isRounded: tableDisable,
  disabled: tableDisable,
  style: tableDisable,
  type: tableDisable,
  isLoading: tableDisable,
  className: tableDisable,
  children: tableDisable,
  icon: tableDisable,
  onlyIcon: tableDisable,
  iconAlignment: tableDisable,
  iconProps: tableDisable,
  size: tableDisable,
  loadingColor: tableDisable,
  onClick: tableDisable,
  href: tableDisable,
  id: tableDisable,
  testId: tableDisable
}

export const WithIcon = (args) => {

  return (
    <>
      <span>
        <Button onClick={action('onClick')} buttonType="primary" iconAlignment={args.iconAlignment} icon={args.icon}>
          {args.label}
        </Button>
      </span>
    </>
  );
};

WithIcon.args = { label: 'Publish', icon: 'MarketplaceSmallFilledWhite' };
WithIcon.parameters = {
  docs: {
    storyDescription: 'You can provide icons to buttons as follows:'
  }
}
WithIcon.argTypes = {
  isFullWidth: tableDisable,
  isRounded: tableDisable,
  disabled: tableDisable,
  style: tableDisable,
  type: tableDisable,
  isLoading: tableDisable,
  className: tableDisable,
  buttonType: tableDisable,
  children: tableDisable,
  onlyIcon: tableDisable,
  iconProps: tableDisable,
  size: tableDisable,
  loadingColor: tableDisable,
  onClick: tableDisable,
  href: tableDisable,
  id: tableDisable,
  testId: tableDisable
}

export const Disabled = (args) => (
  <Button onClick={action('onClick')} disabled={args.disabled}>
    {args.label}
  </Button>
);
Disabled.args = { label: 'Disabled', disabled: true };
Disabled.parameters = {
  docs: {
    storyDescription: 'A non-functional button, depending on the use case, can be displayed in a disabled state.'
  }
}
Disabled.argTypes = {
  isFullWidth: tableDisable,
  isRounded: tableDisable,
  style: tableDisable,
  type: tableDisable,
  isLoading: tableDisable,
  className: tableDisable,
  buttonType: tableDisable,
  children: tableDisable,
  icon: tableDisable,
  onlyIcon: tableDisable,
  iconAlignment: tableDisable,
  iconProps: tableDisable,
  size: tableDisable,
  loadingColor: tableDisable,
  onClick: tableDisable,
  href: tableDisable,
  id: tableDisable,
  testId: tableDisable
}

export const WithLoader = (args) => (
  <>
    <Button onClick={action('onClick')} buttonType="primary" icon="PublishAsset" isLoading={args.isLoading}>
      {args.label}
    </Button>
    <Button onClick={action('onClick')} className="ml-10" buttonType="secondary" icon="Save" isLoading={args.isLoading} loadingColor={args.loadingColor}>
      {args.label}
    </Button>
  </>
);
WithLoader.args = { label: 'Publish', isLoading: true, loadingColor: "#6c5ce7" };
WithLoader.parameters = {
  docs: {
    storyDescription: 'You can provide a loading state to buttons until its respective actions are performed.'
  }
};
WithLoader.argTypes = {
  isFullWidth: tableDisable,
  isRounded: tableDisable,
  style: tableDisable,
  icon: tableDisable,
  type: tableDisable,
  className: tableDisable,
  buttonType: tableDisable,
  children: tableDisable,
  disabled: tableDisable,
  onlyIcon: tableDisable,
  iconAlignment: tableDisable,
  iconProps: tableDisable,
  size: tableDisable,
  onClick: tableDisable,
  href: tableDisable,
  id: tableDisable,
  testId: tableDisable
}

