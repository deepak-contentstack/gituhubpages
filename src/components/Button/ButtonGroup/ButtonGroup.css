@import '../../../../tokens/variables.scss';
@import '../../../../tokens/tokens.css';
@import '../../../static/styles/css/typography.css';
@import '../../../static/styles/css/variable.css';
@import '../../../static/styles/css/common.css';

.Button__group {
    button {
        margin-right: $space-20;
        &:last-child {
            margin-right: 0;
        }
    }
}
.ui-component .button-group {
  	margin: 0 calc(var(--space-2) + 1px);
}