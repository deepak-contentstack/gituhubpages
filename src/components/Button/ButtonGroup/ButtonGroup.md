# ButtonGroup

## Best practices

- Provide adequate spacings when you group two or more buttons.
- Use style prop to a group of buttons wrapped in ButtonGroup component to add different styles as per requirement.
