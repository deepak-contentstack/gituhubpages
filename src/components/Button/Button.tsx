import React, {
  Component,
  CSSProperties,
  MouseEvent as ReactMouseEvent,
  MouseEventHandler,
  Fragment,
} from 'react'
import cn from 'classnames'

import withDeprecatedProp from '../../utils/hooks/depricatedPropsHoc';

// import Icon, { IconProps } from '../Icon2'
// import { iconName } from '../Icon2/constants'
// import AsyncLoader from '../AsyncLoader/AsyncLoader'

import './Button.css'

const defaultProps = {
  isFullWidth: false,
  isRounded: false,
  disabled: false,
  buttonType: 'primary',
  type: 'button',
  isLoading: false,
  testId: 'cs-button'
}

type DefaultProps = Readonly<typeof defaultProps>

export type ButtonProps = {
  /**
   * Class names to be appended to the className prop of the Button Component
   */
  className?: string
  children?: React.ReactNode
  /**
   * Pass the icon that want to add in button
   */
  icon?: any
  /**
   * If you want to show only icon, set the param to ‘true’
   */
  onlyIcon?: boolean
  /**
   * Determines the alignment of the icon
   */
  iconAlignment?: 'left' | 'right'
  /**
   * Pass additional props for icon
   */
  // iconProps?: Partial<IconProps>
  /**
   * Pass an id to the button
   */
  id?: string
  /**
   * Determine the type of button you want to use
   */
  buttonType?:
  | 'primary'
  | 'secondary'
  | 'tertiary'
  | 'tertiary-outline'
  | 'outline'
  | 'success'
  | 'danger'
  | 'warning'
  | 'light'
  | 'control'
  | 'delete'
  | 'bulk-delete'
  | 'outline-delete'
  | 'outline-success'
  | 'white'
  /**
   * Determine the size of the button: small or large
   */
  size?: 'small' | 'large'
  /**
   * If button covers full width
   */
  isFullWidth?: boolean
  /**
   * If button is rounded
   */
  isRounded?: boolean
  /**
   * An ID used for testing purposes applied as a data attribute (data-test-id)
   */
  testId?: string
  /**
   * Used to set the button in the loading state
   */
  isLoading?: boolean
  /**
   * Determines the color of the loading animation you want to display
   */
  loadingColor?: string
  /**
   * Action that you want to perform when a button is clicked
   */
  onClick?: MouseEventHandler
  /**
   * Determines whether button is disabled or not
   */
  disabled?: boolean
  /**
   * If you want a button as a link, provide the details in href.
   */
  href?: string
  /**
   * Type of button action (button, submit,reset)
   */
  type?: 'button' | 'submit' | 'reset'
  /**
   * Pass any CSS properties for the button
   */
  style?: CSSProperties
} & Partial<DefaultProps>
export class Button extends Component<ButtonProps> {
  public static propTypes: any = {}
  static defaultProps = defaultProps
  render() {
    const {
      className,
      children,
      buttonType,
      size,
      isFullWidth,
      isRounded,
      onClick,
      disabled,
      href,
      type,
      icon,
      iconAlignment,
      style,
      isLoading,
      loadingColor,
      onlyIcon,
      testId,
      id,
      iconProps = {},
      ...otherProps
    }: any = this.props
    const classNames = cn('Button', className, [`Button--${buttonType}`], {
      [`Button--state-disabled`]: disabled,
      [`Button--size-${size}`]: size,
      ['Button--style-fullwidth']: isFullWidth,
      ['Button--style-circle']: isRounded,
      [`Button--icon-alignment-${iconAlignment}`]: iconAlignment,
      [`Button--state-loading`]: isLoading,
      [`Button--icon`]: onlyIcon,
    })
    const addButtonIconClassNames = cn('Button__icon')
    const Element = href ? 'a' : 'button'
    return (
      <Element
        data-test-id={testId}
        className={classNames}
        disabled={disabled}
        href={!disabled ? href : null}
        id={id}
        aria-label={id || 'aria-button'}
        type={type}
        style={style}
        onClick={(e: any) => {
          if (onClick) {
            onClick(e)
          }
        }}
        {...otherProps}
      >
        <div className="flex-center">
          {isLoading &&
            <div className="Button__loader">
              {/* <AsyncLoader color={loadingColor} /> */}
            </div>
          }
          <div
            style={{ visibility: !isLoading ? 'visible' : 'hidden' }}
            className={iconAlignment === "right" ? "flex-right" : "flex-v-center"}
          >
            {buttonType === 'control' && (
              <span className={addButtonIconClassNames}>
                {/* <Icon icon={iconName.PlusSign} /> */}
              </span>
            )}
            {/* {icon && <Icon icon={icon} {...iconProps} />} */}
            {children && <>{children}</>}
          </div>
        </div>
      </Element>
    )
  }
}

export default withDeprecatedProp(Button, { 'loading': 'isLoading' })