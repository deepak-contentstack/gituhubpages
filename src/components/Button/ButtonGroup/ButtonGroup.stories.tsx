import React from 'react'
import ButtonGroup from './ButtonGroup'
import notes from './ButtonGroup.md'
import { action } from '@storybook/addon-actions';

import { Button } from '../Button'
import DocSubData from '../../../utils/components/DocSubData'

const componentDescription = 'ButtonGroup component can be used to group two or more buttons together.'
const usageDescription='You can group two or more Buttons in a ButtonGroup when an action required by the user has more than one option. Here, the negative action button is paired with a positive action button in that order. The negative action buttons are placed on the left, whereas the positive action buttons should be placed on the right.'

export default {
    title: 'Components /Button/ButtonGroup',
    component: ButtonGroup,
    argTypes:{
        className: {
        control: { type: 'disabled' },
        description: 'Provide the class names to be appended to this prop',
        table: {
            type: {
            summary: 'string'
            }
        }
        },
        children: {
        control: { type: 'disabled' },
        description: 'Add child elements defined within a component.',
        table: {
            type: {
            summary: 'ReactReactNode'
            }
        }
        },
        style: {
        control: { type: 'disabled' },
        description: 'Pass the CSS properties for the button group.',
        table: {
            type: {
            summary: 'CSSProperties'
            }
        }
        },
        testId: {
        control: { type: 'diabled' },
        description: 'Pass an ID that you can use for testing purposes. It is applied as a data attribute (data-test-id).',
        table: {
            type: {
            summary: 'string'
            },
            defaultValue: {
            summary: 'cs-button-group'
            }
        },
        }
    },
    parameters: {
        componentSubtitle: <DocSubData componentDescription={componentDescription} usageDescription={usageDescription} />,
        notes: { notes },
        design: {
            type: 'figma',
            url:
              'https://www.figma.com/proto/NUaPd7QxttcJAQfx9KYjBu/01-%E2%80%A2-Atoms-%26-Properties?page-id=52%3A6&node-id=4352%3A13557&viewport=241%2C48%2C0.12&scaling=min-zoom',
          }
    }
}

const tableDisable = { table: { disable: true } }

export const Default = (args) => (
    <ButtonGroup>
      <Button onClick={action('onClick')} buttonType="light">Cancel</Button>
      <Button onClick={action('onClick')} buttonType="secondary" icon="UnpublishAsset">Unpublish</Button>
      <Button onClick={action('onClick')} buttonType="primary" icon="PublishWhite">Publish</Button>
    </ButtonGroup>
  )
  
