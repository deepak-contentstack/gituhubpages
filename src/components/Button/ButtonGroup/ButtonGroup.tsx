import React, {CSSProperties} from 'react'
import cn from 'classnames'
import './ButtonGroup.css'

export type ButtonGroupProps = {
  /**
   * Provides the class names to be appended to this prop.
   */
  className?:string,
  /**
   * Add child elements defined within a component.
   */
  children?: React.ReactNode
  /**
   * Pass the CSS properties for the button group.
   */
  style?:CSSProperties,
  /**
   * Pass an ID that you can use for testing purposes. It is applied as a data attribute (data-test-id).
   */
  testId?: string
} 

const ButtonGroup = (props: ButtonGroupProps) => {
  const {className,children,style,testId,...otherProps}: any =props
  const classNames = cn('Button__group', className)

  return (
    <div className={classNames} data-test-id={testId} {...otherProps}>
      {children && children}
    </div>
  )
  }

  ButtonGroup.defaultProps = {
    testId: 'cs-button-group'
  } as Partial<ButtonGroupProps>;

export default ButtonGroup;