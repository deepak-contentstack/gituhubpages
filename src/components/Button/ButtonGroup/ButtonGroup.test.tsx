import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import ButtonGroup from './ButtonGroup'
import Button from '../Button'

describe('ButtonGroup', () => {
it('renders ButtonGroup component', () => {
    const output = renderer.create(
    <ButtonGroup>
    <Button  buttonType="light">Cancel</Button>
    <Button  buttonType="secondary" icon="UnpublishAsset">Unpublish</Button>
    <Button  buttonType="primary" icon="PublishWhite">Publish</Button>
    </ButtonGroup>
    ).toJSON();

    expect(output).toMatchSnapshot();
});

it('ButtonGroup with classname', () => {
    const element = mount(
    <ButtonGroup testId="custom-test-id" className="test-classname">
    <Button  buttonType="light">Cancel</Button>
    <Button  buttonType="secondary" icon="UnpublishAsset">Unpublish</Button>
    <Button  buttonType="primary" icon="PublishWhite">Publish</Button>  
    </ButtonGroup>);
    expect(element.find('[data-test-id="custom-test-id"]').props().className.includes('test-classname')).toBe(true);
});

it('It should render a ButtonGroup with custom style', () => {
const CustomStyle = () => {
    return (
    <ButtonGroup style={{ color: 'red' }}>
    <Button  buttonType="light">Cancel</Button>
    <Button  buttonType="secondary" icon="UnpublishAsset">Unpublish</Button>
    <Button  buttonType="primary" icon="PublishWhite">Publish</Button>
    </ButtonGroup>
    )
};
const tree = renderer.create(<CustomStyle />).toJSON()
expect(tree).toMatchSnapshot();
});

})