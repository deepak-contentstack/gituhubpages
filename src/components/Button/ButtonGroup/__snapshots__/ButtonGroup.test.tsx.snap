// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup It should render a ButtonGroup with custom style 1`] = `
<div
  className="Button__group"
  data-test-id="cs-button-group"
>
  <button
    aria-label="aria-button"
    className="Button Button--light"
    data-test-id="cs-button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <div
      className="flex-center"
    >
      <div
        className="flex-v-center"
        style={
          Object {
            "visibility": "visible",
          }
        }
      >
        Cancel
      </div>
    </div>
  </button>
  <button
    aria-label="aria-button"
    className="Button Button--secondary"
    data-test-id="cs-button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <div
      className="flex-center"
    >
      <div
        className="flex-v-center"
        style={
          Object {
            "visibility": "visible",
          }
        }
      >
        Unpublish
      </div>
    </div>
  </button>
  <button
    aria-label="aria-button"
    className="Button Button--primary"
    data-test-id="cs-button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <div
      className="flex-center"
    >
      <div
        className="flex-v-center"
        style={
          Object {
            "visibility": "visible",
          }
        }
      >
        Publish
      </div>
    </div>
  </button>
</div>
`;

exports[`ButtonGroup renders ButtonGroup component 1`] = `
<div
  className="Button__group"
  data-test-id="cs-button-group"
>
  <button
    aria-label="aria-button"
    className="Button Button--light"
    data-test-id="cs-button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <div
      className="flex-center"
    >
      <div
        className="flex-v-center"
        style={
          Object {
            "visibility": "visible",
          }
        }
      >
        Cancel
      </div>
    </div>
  </button>
  <button
    aria-label="aria-button"
    className="Button Button--secondary"
    data-test-id="cs-button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <div
      className="flex-center"
    >
      <div
        className="flex-v-center"
        style={
          Object {
            "visibility": "visible",
          }
        }
      >
        Unpublish
      </div>
    </div>
  </button>
  <button
    aria-label="aria-button"
    className="Button Button--primary"
    data-test-id="cs-button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <div
      className="flex-center"
    >
      <div
        className="flex-v-center"
        style={
          Object {
            "visibility": "visible",
          }
        }
      >
        Publish
      </div>
    </div>
  </button>
</div>
`;
